tests/e2e/framework/testInfo.go

```diff
@@ -111,6 +123,102 @@ func (t testInfo) Update(r int) error {
 	return t.createStatusFile(r)
 }
 
+func (t testInfo) FetchAndSaveClusterLogs() error {
+	if *projectID == "" {
+		return nil
+	}
+	// connect to stackdriver
+	ctx := context.Background()
+	glog.Info("Fetching cluster logs")
+	loggingClient, err := logging.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	fetchAndWrite := func(logName string) error {
+		glog.Info(fmt.Sprintf("Fetching logs on %s\n", logName))
+		// fetch logs from pods created for this run only
+		filter := fmt.Sprintf(
+			`logName = "%s" AND
+			resource.labels.namespace_id = "%s"`,
+			logName, *namespace)
```

Comment:
I too agree that dereferencing *namespace directly creates confusion on the code structure as it may be unclear where this value is initialized. However, I think passing namespace as a param to FetchAndSaveClusterLogs() is better than declaring as member testInfo. For one, namespace is already a member (read from command-line flag) of KubeInfo, and calling newKubeInfo requires a testInfo, so defining namespace in testInfo creates a cyclic dependency. Also, logically, namespace is a property of kubectl and less so for testInfo.